RHDEVDOCS-6267

deerskindoll opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.
Do not create or rename a top-level directory (or any subdirectory in a directory that contains a hugebook.flag file) in the repository and topic map without checking with a docs program manager first.
If a book is being created or modified, there are changes on the Customer Portal that must also be made.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

reference modules for optional pod configuration fields in `Build` and `BuildRun` CRs

Version(s):
<!--- Specify the version or versions of OpenShift your PR applies to. -->

build-docs-main
build-docs-1.4

Issue:
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

https://issues.redhat.com/browse/RHDEVDOCS-6267

Link to docs preview:
- https://92126--ocpdocs-pr.netlify.app/openshift-builds/latest/configuring/configuring-build-runs.html
- https://92126--ocpdocs-pr.netlify.app/openshift-builds/latest/configuring/configuring-openshift-builds.html
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

no QE available

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

deerskindoll commented:
@sayan-biswas @adambkaplan 

I apologize for submitting this so late

deerskindoll commented:
@sayan-biswas @adambkaplan 

please review and approve this PR if everything is ok.

deerskindoll commented:
/label peer-review-needed

lahinson commented on configuring/configuring-build-runs.adoc:
```diff
@@ -7,9 +7,13 @@ include::_attributes/common-attributes.adoc[]
 
 
-In a `BuildRun` custom resource (CR), you can define the build reference, build specification, parameter values, service account, output, retention parameters, and volumes to configure a build run. A `BuildRun` resource is available for use within a namespace.
+In a `BuildRun` custom resource (CR), you can define the build reference, build specification,
```
```suggestion
In a `BuildRun` custom resource (CR), you can configure a build run by defining the build reference, build specification, parameter values, service account, output, retention parameters, and volumes to configure a build run. You can also configure build pods by using a `Build` CR.
```
From the [OCP doc guidelines](https://92126--ocpdocs-pr.netlify.app/openshift-builds/latest/configuring/configuring-build-runs.html
https://92126--ocpdocs-pr.netlify.app/openshift-builds/latest/configuring/configuring-openshift-builds.html): "Do not hard wrap lines at 80 characters (or at any other length)."

lahinson commented on configuring/configuring-build-runs.adoc:
```diff
@@ -7,9 +7,13 @@ include::_attributes/common-attributes.adoc[]
 
-In a `BuildRun` custom resource (CR), you can define the build reference, build specification, parameter values, service account, output, retention parameters, and volumes to configure a build run. A `BuildRun` resource is available for use within a namespace.
+In a `BuildRun` custom resource (CR), you can define the build reference, build specification,
+parameter values, service account, output, retention parameters, and volumes to configure a build run.
```
```suggestion

```
If you implement my first suggestion, you can remove this line.

lahinson commented on configuring/configuring-build-runs.adoc:
```diff
@@ -7,9 +7,13 @@ include::_attributes/common-attributes.adoc[]
-In a `BuildRun` custom resource (CR), you can define the build reference, build specification, parameter values, service account, output, retention parameters, and volumes to configure a build run. A `BuildRun` resource is available for use within a namespace.
+In a `BuildRun` custom resource (CR), you can define the build reference, build specification,
+parameter values, service account, output, retention parameters, and volumes to configure a build run.
+You can also use `Build` CR to configure build pods.
```
```suggestion

```
If you implement my first suggestion, you can remove this line.

lahinson commented on configuring/configuring-openshift-builds.adoc:
```diff
@@ -7,9 +7,13 @@ include::_attributes/common-attributes.adoc[]
 
 
-In a `Build` custom resource (CR), you can define the source, build strategy, parameter values, output, retention parameters, and volumes to configure a build. A `Build` resource is available for use within a namespace.
+In a `Build` custom resource (CR), you can define the source, build strategy,
```
```suggestion
In a `Build` custom resource (CR), you can configure a build by defining the source, build strategy, parameter values, output, retention parameters, and volumes. You can also configure build pods by using a `Build` CR.
```
Similar edits here as what I suggested in `configuring-build-runs.adoc`. See my comments there.

lahinson commented on configuring/configuring-openshift-builds.adoc:
```diff
@@ -7,9 +7,13 @@ include::_attributes/common-attributes.adoc[]
 
-In a `Build` custom resource (CR), you can define the source, build strategy, parameter values, output, retention parameters, and volumes to configure a build. A `Build` resource is available for use within a namespace.
+In a `Build` custom resource (CR), you can define the source, build strategy,
+parameter values, output, retention parameters, and volumes to configure a build.
```
```suggestion

```

lahinson commented on configuring/configuring-openshift-builds.adoc:
```diff
@@ -7,9 +7,13 @@ include::_attributes/common-attributes.adoc[]
-In a `Build` custom resource (CR), you can define the source, build strategy, parameter values, output, retention parameters, and volumes to configure a build. A `Build` resource is available for use within a namespace.
+In a `Build` custom resource (CR), you can define the source, build strategy,
+parameter values, output, retention parameters, and volumes to configure a build.
+You can also use `Build` CR to configure build pods.
```
```suggestion

```

lahinson commented on modules/ob-configuring-pods-in-build-run.adoc:
```diff
@@ -0,0 +1,21 @@
+
+You can use the following optional `BuildRun` CR fields to configure {builds-shortname} pods:
+
+* The `spec.tolerations` field specifies the tolerations for the {builds-shortname} pod with the following limitations:
```
```suggestion
* The `spec.tolerations` field specifies the tolerations for the {builds-shortname} pod with the following limitation:
```
It looks like there is only one limitation.

lahinson commented on modules/ob-configuring-pods-in-build-run.adoc:
```diff
@@ -0,0 +1,21 @@
+
+* The `spec.tolerations` field specifies the tolerations for the {builds-shortname} pod with the following limitations:
++
+** Only the `NoSchedule` taint `effect` is supported
```
```suggestion
** Only the `NoSchedule` taint `effect` is supported.
```
If possible, add whatever the relevant noun is after `effect`. See "Object references" in the OCP docs guidelines for more information: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references

lahinson commented on modules/ob-configuring-pods-in-build-run.adoc:
```diff
@@ -0,0 +1,21 @@
+[id="ob-configuring-pods-in-build-run_{context}"]
+= Pod configuration
+
+You can use the following optional `BuildRun` CR fields to configure {builds-shortname} pods:
```
In the preview, I see that `{builds-shortname} pods` renders as `Builds pods`. Is that correct, or should it be `Build pods`? I ask because in the assembly files in this PR, I see the phrase `build pods` used.

deerskindoll replied:
it should be Builds pods because it refers to pods created with the product (aka Builds).

lahinson commented on modules/ob-configuring-pods-in-build-run.adoc:
```diff
@@ -0,0 +1,21 @@
+** Only the `NoSchedule` taint `effect` is supported
++
+* The `spec.nodeSelector` field specifies which nodes the {builds-shortname} pod should run on.
+* The `spec.schedulerName` field specifies the scheduler for the {builds-shortname} pod
```
```suggestion
* The `spec.schedulerName` field specifies the scheduler for the {builds-shortname} pod.
```

lahinson commented on modules/ob-configuring-pods-in-build.adoc:
```diff
@@ -0,0 +1,22 @@
+[id="ob-configuring-pods-in-build_{context}"]
+= Pod configuration
+
+You can use the following optional `Build` CR fields to configure {builds-shortname} pods:
```
See my earlier comment about `{builds-shortname} pods` vs. `Build pods`. Should the phrase be `Builds pods`, `Build pods`, or `build pods`?

lahinson commented on modules/ob-configuring-pods-in-build.adoc:
```diff
@@ -0,0 +1,22 @@
+
+You can use the following optional `Build` CR fields to configure {builds-shortname} pods:
+
+* The `spec.tolerations` field specifies the tolerations for the {builds-shortname} pod with the following limitations:
```
```suggestion
* The `spec.tolerations` field specifies the tolerations for the {builds-shortname} pod with the following limitation:
```
Only one limitation is listed.

lahinson commented on modules/ob-configuring-pods-in-build.adoc:
```diff
@@ -0,0 +1,22 @@
+
+* The `spec.tolerations` field specifies the tolerations for the {builds-shortname} pod with the following limitations:
++
+** Only the `NoSchedule` taint `effect` is supported
```
```suggestion
** Only the `NoSchedule` taint `effect` is supported.
```
If possible, include the correct object reference for `effect`. See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references.

lahinson commented on modules/ob-configuring-pods-in-build.adoc:
```diff
@@ -0,0 +1,22 @@
+** Only the `NoSchedule` taint `effect` is supported
++
+* The `spec.nodeSelector` field specifies which nodes the {builds-shortname} pod should run on.
+* The `spec.schedulerName` field specifies the scheduler for the {builds-shortname} pod
```
```suggestion
* The `spec.schedulerName` field specifies the scheduler for the {builds-shortname} pod.
```

lahinson reviewed (commented):
Nice work! I provided some suggestions based on the OCP docs guidelines for your consideration. I wasn't able to use the generated preview to confirm that the table formatting was correct, so you might want to double check that locally. Before you submit this for merge review, be sure to squash your commits down to 1. Thanks!

mramendi committed:
RHDEVDOCS 6426 Remove language note + RHDEVDOCS 6267

RHDEVDOCS-6267

added 1.2.1 RN file

Remove prerequisite from builds

fixed example file

moved callout

replaced `shp` CLI download link

RHDEVDOCS-6267

RHDEVDOCS-6267

fixed example file

moved callout

replaced `shp` CLI download link

RHDEVDOCS-6267

RHDEVDOCS-6267

Update modules/ob-configuring-pods-in-build-run.adoc

Co-authored-by: Laura Hinson <lahinson@redhat.com>

Update configuring/configuring-build-runs.adoc

Co-authored-by: Laura Hinson <lahinson@redhat.com>

Update configuring/configuring-build-runs.adoc

Co-authored-by: Laura Hinson <lahinson@redhat.com>

Update configuring/configuring-openshift-builds.adoc

Co-authored-by: Laura Hinson <lahinson@redhat.com>

Update configuring/configuring-build-runs.adoc

Co-authored-by: Laura Hinson <lahinson@redhat.com>

Update configuring/configuring-openshift-builds.adoc

Co-authored-by: Laura Hinson <lahinson@redhat.com>

Update modules/ob-configuring-pods-in-build-run.adoc

Co-authored-by: Laura Hinson <lahinson@redhat.com>

Update configuring/configuring-openshift-builds.adoc

Co-authored-by: Laura Hinson <lahinson@redhat.com>

Update modules/ob-configuring-pods-in-build-run.adoc

Co-authored-by: Laura Hinson <lahinson@redhat.com>

Update modules/ob-configuring-pods-in-build.adoc

Co-authored-by: Laura Hinson <lahinson@redhat.com>

Update modules/ob-configuring-pods-in-build.adoc

Co-authored-by: Laura Hinson <lahinson@redhat.com>

Update modules/ob-configuring-pods-in-build.adoc

Co-authored-by: Laura Hinson <lahinson@redhat.com>

fixed tables

fixed formatting

deerskindoll commented:
/label merge-review-needed

gabriel-rh reviewed (approved):
/LGTM

gabriel-rh commented:
/cherrypick build-docs-1.4
